internal/terraform/context_apply_deferred_test.go

```diff
@@ -1721,19 +1735,24 @@ output "a" {
 				},
 				inputs:      map[string]cty.Value{},
 				wantPlanned: map[string]cty.Value{
-					// The all resources will be deferred, so shouldn't
-					// have any action at this stage.
+					// Empty because it's a refresh-only plan in this stage.
 				},
 
 				wantActions: map[string]plans.Action{},
 				wantApplied: map[string]cty.Value{
-					// The all resources will be deferred, so shouldn't
+					// The resource will be deferred, so shouldn't
 					// have any action at this stage.
 				},
 				wantOutputs: map[string]cty.Value{
+					// TODO NF: In both the refresh-only stage and the real-plan stage below, the output is
+					// getting the new `name` value from the partially-planned deferred change rather than the
+					// old `name` value from the state. Is that right? For both cases? I couldn't quite
+					// decide, but currently that's what happens because of where the early-out happens over
+					// in GetResource while we're building the hcl eval context.
 					"a": cty.ObjectVal(map[string]cty.Value{
```

Comment:
In the other tests we always return null values? Is it possible to do the same here? Or is this simply because there is something already in the state for this? Maybe we could return an unknown value here instead of null? To represent that something may be changing but we're not sure.